packages/kbn-interpreter/src/public/interpreter.js

```diff
@@ -42,29 +42,34 @@ export async function initializeInterpreter(socket, typesRegistry, functionsRegi
     return interpretFn(ast, context);
   };
 
-  // Listen for interpreter runs
-  socket.on('run', ({ ast, context, id }) => {
-    const types = typesRegistry.toJS();
-    const { serialize, deserialize } = serializeProvider(types);
-    interpretAst(ast, deserialize(context)).then(value => {
-      socket.emit(`resp:${id}`, { value: serialize(value) });
+  if (socket) {
```

Comment:
if we don't have a connected socket, ignore this part (server functions won't be available)

Reply:
`socket` is also used above, passed into `socketInterpreterProvider`.  Figuring out this code may take me some time, I'm pretty confused by the flow.  Seems like just ripping sockets out of here would help immensely with following the logic.

@chrisdavies, in https://github.com/elastic/kibana/issues/29118, will removing sockets be step one with the other steps to follow, or do they have to happen in unison?  I wonder if we could remove sockets in time for for 7.0.

Reply:
socketInterpreterProvider will only use socket if it knows there is a function on the other side it can use.
we resolve here with `[]` if socket is not available meaning there are no functions on the other side

Reply:
i taught the plan was to work on removing sockets, but as that is highly unrealistic to happen by tuesday, we also do this. This will make kibana work with interpreter even if socket couldn't be established, as all our functions are clientside

Reply:
and its super small change